source/common/grpc/async_client_impl.cc

```diff
@@ -229,6 +267,7 @@ void AsyncStreamImpl::sendMessageRaw(Buffer::InstancePtr&& buffer, bool end_stre
 void AsyncStreamImpl::closeStream() {
   Buffer::OwnedImpl empty_buffer;
   stream_->sendData(empty_buffer, true);
+  current_span_->finishSpan();
```

Comment:
Should we `setTag` value here before finishSpan? 
This is local close path. IIUC,  similarly  `Canceled` is used in asyncRequest path 
+@wbpcode  who is trace expert , WDYT?

(Also, please see my comment below on how to test this path)

Reply:
Makes sense. I added the tag and the test as suggested.